### DISCUSSION TITLE
Using public (own) TLS for Teleport  in a kubernetes cluster

### DISCUSSION THREAD

**Discussion by erickessels** [gravitational/teleport#9598] [Q&A] [answered]
I have installed a teleport cluster in kubernetes with the helm chart teleport/teleport-cluster witch works fine. I would like to use our public certficate for the proxy but can not find the instruction for it. I tried to use the "old" way by creating a secret named tls-web but it does not pick that up this secret during installation.

Would love to get some guidance here.

Thanks, Eric

**Comment by dmitry-mightydevops** [comment]
I have similar issue. I want to use teleport.site.com behind cloudflare proxy dns. Which (in cloudflare strict mode) requires installation of a origin server cert: https://developers.cloudflare.com/ssl/origin-configuration/origin-ca

so it's not clear how to pass the name of the tls secret holding that cert/pem files in 
https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-cluster/templates/deployment.yaml

I can use cloudflare in DNS only mode and then things do work, but I want teleport be behind cloudflare (in proxy mode). 

Please advise.

**Comment by dmitry-mightydevops** [comment]
It seems there is an outstanding PR to achieve that functionality
https://github.com/gravitational/teleport/pull/8372

but it didn't get merged :(

**Comment by dmitry-mightydevops** [comment]
```
# extraEnv sets additional environment variables inside the chart's container
# extraEnv:
#  - name: SSL_CERT_DIR
#    value: /etc/teleport-tls:/etc/ssl/certs
extraEnv: []
```

so it seems from this error message (teleport cli):


```
If you have your own certificate authority that you would like to use to
  validate the certificate chain presented by the proxy, set the
  SSL_CERT_FILE and SSL_CERT_DIR environment variables respectively and try
  again.
```

things should work by passing these 2 values. What's not clear is what file should be in this **SSL_CERT_DIR** (i.e. what names)

**Comment by dmitry-mightydevops** [comment]
It seems old chart had this functionality (however the chart still receives updates, but still is marked as deprecated)
https://github.com/gravitational/teleport/blob/2ddf2411e15502b7bbce70ca01a6ba43a4af4fb2/examples/chart/teleport/templates/deployment.yaml#L71

**Reply by erickessels** [reply]
I had the same impression that that one is deprecated, also discovered that in custom chart mode the use of the cert-manager is not supported, as specified in the docs. I have configured that but still testing. Can you use it in chartmode custom ?

**Reply by erickessels** [reply]
```
helm install teleport teleport/teleport-cluster \
  --namespace teleport \
  --set chartMode=custom \
  --set clusterName=teleport.x.x.x \
  --set podSecurityPolicy.enabled=false \
  --set highAvailability.replicaCount=2 \
  --set highAvailability.podDisruptionBudget.enabled=true \
  --set highAvailability.podDisruptionBudget.minAvailable=1 \
  --set highAvailability.certManager.enabled=true \
  --set highAvailability.certManager.issuerName=letsencrypt
```

**Reply by dmitry-mightydevops** [reply]
I have never used `--set chartMode=custom`

**Comment by webvictim** [comment]
As an FYI for anyone with this issue, I've taken a crack at adding support for a custom TLS secret in this PR: https://github.com/gravitational/teleport/pull/11295

Testing welcome!

**Comment by webvictim** [comment, marked as answer]
This functionality was merged and you can now provide your own certs/CA using these values: https://github.com/gravitational/teleport/blob/339bc6f25bcf2c099ff826b899fc7affa5688b02/examples/chart/teleport-cluster/values.yaml#L172-L184
